README.md

```diff
@@ -263,81 +280,110 @@ collection = ParentEventsTreeCollectionProvider5(events, data_source=data_source
 collection.show()
 ```
 
-## 2.3. Theory
+## 2.3. Short theory
 
-The library provides stream data and some tools for data manipulation.
-
-What’s the definition of a stream?   
-A short definition is "a sequence of elements from a source that supports aggregate operations."
+The library provides tools for handling stream data. What’s a stream? It's a sequence of elements from a source that
+supports aggregate operations.
 
 ### Terms
 
-- **Data object**: An object of `Data` class which is wrapper under stream. 
+- **Data object**: An instance of `Data` class which is wrapper under stream.
+- **Sequence of elements**:
+  A _Data object_ provides an interface to a sequenced set of values of a specific element type. Stream inside the _Data
+  object_ **don’t actually store** elements; they are computed on demand.
+- **DataSource**:
+  Any source of data. E.g. [Report Data Provider](https://github.com/th2-net/th2-rpt-data-provider), collections,
+  arrays, or I/O resources.
+- **ProviderDataSource**:
+  DataSource object the source of which is [Report Data Provider](https://github.com/th2-net/th2-rpt-data-provider).
+- **SourceAPI**:
+  Each source has its own API to retrieve data. That term/class is used for reference/describe this API.
```

Comment:
I'm a bit confused. What do you mean about this sentence ?

Reply:
Is it better?

The DataSource object whose source is [Report Data Provider](https://github.com/th2-net/th2-rpt-data-provider).
----
Each source has its own API to retrieve data. SourceAPI is a class that provide API for some data source.

Reply:
Yes, it is